### ISSUE TITLE
Eliminating redundant function parameter "$__unused"

### ISSUE THREAD

**Issue by thautwarm** [purescript/purescript#4201]
https://github.com/purescript/purescript/blob/c14d52e0bd4caca88358ac082911e8a94d072dd5/src/Language/PureScript/CodeGen/JS.hs#L231

The corefn output contains abstractions that have a parameter `"$__unused"`.

```json
{ // ...
"argument":"$__unused","type":"Abs"}
```

They are not filtered by the `UnusedIdent` pattern, and I solve this by filtering the pattern `"Ident "$__unused"`:

```haskell
let jsArg = case arg of
              UnusedIdent -> []
              Ident "$__unused" -> []
              _           -> [identToDiana arg]
```

This might be helpful to alternative backends, as not all programming languages support omitting function arguments:

```python
def f(__unused):
   pass

f() # TypeError: f() missing 1 required positional argument: '__unused'
```

**Comment by thautwarm** [comment]
Above fix is a downstream fix.  

As `identToJSON` generates `$__unused` from `UnusedIdent`,  `identFromJSON` should parse `UnusedIdent` from `$__unused`. 

https://github.com/purescript/purescript/blob/c14d52e0bd4caca88358ac082911e8a94d072dd5/src/Language/PureScript/CoreFn/FromJSON.hs#L102 


https://github.com/purescript/purescript/blob/c14d52e0bd4caca88358ac082911e8a94d072dd5/src/Language/PureScript/CoreFn/ToJSON.hs#L91


https://github.com/purescript/purescript/blob/c14d52e0bd4caca88358ac082911e8a94d072dd5/lib/purescript-cst/src/Language/PureScript/Names.hs#L85

**Comment by nwolverson** [comment]
Yes, I apparently found the same in the `purerl` backend some time ago:
```
    -- TODO this is mangled in corefn json
    let fixIdent (Ident "$__unused") = UnusedIdent
        fixIdent x = x
```

**Comment by JordanMartinez** [comment]
Closed by #4221
